### ISSUE TITLE
calciteInternalStepperItemChange event sometimes emits when programmatically changing items

### ISSUE THREAD

**Issue by benelan** [Esri/calcite-components#4595]
### Actual Behavior 
`calciteInternalStepperItemChange` inconsistently emits when programmatically changing stepper items. 

### Expected Behavior
`calciteInternalStepperItemChange` should not ever emit when stepper items are changed programmatically. #4398

### Reproduction Sample

https://codepen.io/benesri/pen/NWyjdxB?editors=1011

### Reproduction Steps

1. Click "Next Step" a couple times and notice the position is logged in the console
2. Click "Previous Step" a couple times and notice the position is not logged in the console
3. Repeat step 1 and nothing is logged in the console

### Reproduction Version

next.473

### Regression?

**Comment by benelan** [comment]
Nevermind, it got renamed to `calciteInternalStepperItemChange`

**Comment by benelan** [comment]
Reopening/repurposing this issue because `calciteInternalStepperItemChange` sometimes emits when programmatically changing stepper items, which shouldn't happen after #4398. I updated the issue body

**Comment by driskull** [comment]
I think this would require a refactor because the `stepper-item` is listening for the event on the body in order to do some updates.

**Comment by benelan** [comment]
The action items for this issue are
-  make sure `calciteInternalStepperItemChange` only emits on user interaction
- rename the event to `calciteInternalStepperItemSelect` and make it public

**Comment by geospatialem** [comment]
Verified on `next.500`

https://user-images.githubusercontent.com/5023024/175562221-3f621f83-9575-4bde-9a5c-07462c8bc895.mp4

`

**Comment by kevindoshier** [comment]
This still doesnt seem correct,  in the sample programatically a nextStep, prevStep are being used, but there is the case where we dont use that and the user will just click one of the stepper positions, for that case we do not get back a position, here is a codePen

https://codepen.io/kevindoshier/pen/wvyNLJV

**Comment by driskull** [comment]
@kevindoshier can you create a new issue for this? I have a fix in progress.

**Comment by geospatialem** [comment]
> This still doesnt seem correct, in the sample programatically a nextStep, prevStep are being used, but there is the case where we dont use that and the user will just click one of the stepper positions, for that case we do not get back a position, here is a codePen

@kevindoshier, For some reason the Codepen sample wasn't providing a value for the second or the last stepper item. 

[Updated the html in this sample](https://codepen.io/geospatialem/pen/vYRydaM) that seems to recognize click events. In the updated Codepen are you experiencing the same behavior? If so, can you provide the steps to replicate the experience?

**Comment by kevindoshier** [comment]
I see, we dont have any content within each stepper item for our use case, but that does seem to fix the issue, we use this more for navigation.   Adding empty content to each stepper seems to fix it,  @driskull  should I still add an issue though?

**Comment by driskull** [comment]
@kevindoshier created PR: https://github.com/Esri/calcite-components/pull/4938

**Comment by geospatialem** [comment]
Verified on `next` with https://codepen.io/kevindoshier/pen/wvyNLJV
